src/_pytest/fixtures.py

```diff
@@ -1728,6 +1728,20 @@ def _register_fixture(
         )
 
         faclist = self._arg2fixturedefs.setdefault(name, [])
+
+        by_plugin = fixture_def.baseid == ""
+
+        if not by_plugin and faclist:
+            # If the fixture from a plugin, no conflict detection is performed.
+            if faclist[-1].baseid == fixture_def.baseid:
+                # The same file may create two fixtures with the same name (#12952).
+                msg = (
+                    f"Fixture definition conflict: {name!r} has multiple implementations,"
+                    f"namely {faclist[-1].func!r} and {func}  (from: {nodeid!r})."
+                )
+                print(msg)
```

Comment:
Let's turn this into a warning for now 

We also might want to propose a ruff rule for this case

Reply:
Hi @RonnyPfannschmidt ,  Could you please let me know which type is more appropriate: `PytestWarning` or `PytestDeprecationWarning`?